### ISSUE TITLE
Syntactic sugar: Tree.has_data()

### ISSUE THREAD

**Issue by makukha** [lark-parser/lark#1465]
**Suggestion**
Add new method `Tree.has_data(data: str) -> bool` to determine whether there is a token with given data.

**Describe alternatives you've considered**
The same can be achieved with `find_data()` method, but is too verbose:
```python
try:
    next(find_data(data))
except StopIteration:
    has_data = False
else:
    has_data = True
```

**Additional context**
I'm ready to provide a PR if you find this API reasonable.

**Comment by erezsh** [comment]
I can suggest 

```python
has_data = bool(next(tree.find_data(data), False))
```

**Comment by erezsh** [comment]
Another option is:

```python
for _ in tree.find_data(data):
   # do if true
   break
```

I understand having a method would be a bit more ergonomic, but I feel like it's a bit too small to be worth adding.

But I'll leave it open, if more users would like this method, we can add it in.

**Comment by erezsh** [comment]
Yeah. Honestly, I don't know why itertools doesn't have this method:

```python
EMPTY = object()
def iter_is_empty(i):
    return next(i, EMPTY) is EMPTY
```

**Comment by makukha** [comment]
It seems that people will tend to choose readability over performance and use
```python
has_data = bool(tuple(tree.find_data(data)))
```
